File: javaRestAssured/src/test/java/io/swagger/petstore/PetTestsWithFluentAssertions.java
package io.swagger.petstore;

import io.swagger.petstore.asserts.JMessageResponseAssert;
import io.swagger.petstore.asserts.JPetAssert;
import io.swagger.petstore.controllers.JPetController;
import io.swagger.petstore.models.JMessageResponse;
import io.swagger.petstore.models.JPet;
import org.apache.commons.lang3.RandomStringUtils;
import org.junit.Test;

public class PetTestsWithFluentAssertions {

    @Test
    public void addNewPetToStoreTest() {
        JPet testPet = new JPet(null,
                "Pet_" + RandomStringUtils.randomAlphabetic(8), //Pet name
                RandomStringUtils.randomNumeric(8), //Pet ID
                null, null,
                "available"); //Pet status

        JPet petResponse = new JPetController().addNewPet(testPet);

        JPetAssert.assertThat(petResponse).isEqualTo(testPet);
    }

    @Test
    public void deletePetFromStoreTest() {
        JPet testPet = new JPet(null,
                "Pet_" + RandomStringUtils.randomAlphabetic(8), //Pet name
                RandomStringUtils.randomNumeric(8), null, null, //Pet ID
                "available"); //Pet status

        JPetController petAction = new JPetController();

        petAction.addNewPet(testPet);
        petAction.deletePet(testPet);

        JMessageResponse messageResponse = petAction.getPet(testPet);

        JMessageResponseAssert.assertThat(messageResponse).hasMessage("Pet not found");
    }

}

File: kotlinRestAssuredExample/src/test/kotlin/io/swagger/petstore/PetTestsWithFluentAssertions.kt
package io.swagger.petstore

import io.swagger.petstore.controllers.KPetController
import io.swagger.petstore.models.KPet
import org.amshove.kluent.shouldEqual
import org.apache.commons.lang3.RandomStringUtils
import org.junit.Test

class PetTestsWithFluentAssertions {

    @Test
    fun `Add new pet to store`() {
        val testPet = KPet(id = RandomStringUtils.randomNumeric(10),
                name = "Pet_${RandomStringUtils.randomAlphabetic(8)}",
                status = "available")

        val petResponse = KPetController().addNewPet(testPet)

        petResponse shouldEqual testPet
    }

    @Test
    fun `Delete pet from store test`() {
        val testPet = KPet(id = RandomStringUtils.randomNumeric(10),
                name = "Pet_${RandomStringUtils.randomAlphabetic(8)}",
                status = "available")

        KPetController().apply {
            //with instance of KPetController class call following methods:
            addNewPet(testPet)
            deletePet(testPet)

            getPet(testPet).message shouldEqual "Pet not found"
        }
    }

}
